src/main/java/tigase/server/Packet.java

```diff
@@ -598,6 +601,44 @@ public void setPriority(Priority priority) {
 		return processorsIds;
 	}
 
+	/**
+	 * Method returns JID from-address that was authenticated and bound to user session.
+	 *
+	 * This is a helper method that facilitates and improve maintaining correct stanza from when
+	 * communicating between ClientConnectionManager and SessionManager.
+	 *
+	 * This is a temporary solution! In version 9.0, after reviewing the APIs and correcting clustering strategy
+	 * packets incomming in (Client) Connection Managers should already have correct 'from' attribute in stanzas
+	 * set in {@code tigase.server.xmppclient.ClientConnectionManager#processSocketData(tigase.xmpp.XMPPIOService)}.
+	 *
+	 * @return {@code Optional} that may contain user JID if the session was already authorised.
+	 */
+	@Deprecated
+	@TigaseDeprecated(since = "8.2.0", removeIn = "9.0.0")
+	public Optional<JID> getServerAuthorisedStanzaFrom() {
+		return Optional.ofNullable(serverAuthorisedStanzaFrom);
```

Comment:
Wouldn't it be better to have an Optional instance in place? (created in set method)

Reply:
Probably more optimal to avoid excessive allocation, though I read it's discouraged to use Optionals as class members.

Reply:
Could be. That was just an idea.